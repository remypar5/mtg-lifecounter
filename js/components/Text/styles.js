import { StyleSheet } from 'react-native';

import { normalizeFontSize } from '../../utils';
import { COLOR_FOREGROUND, COLOR_BACKGROUND, FONT_SIZE_LARGE, FONT_SIZE, FONT_FAMILY, FONT_SIZE_SMALL, FONT_SIZE_HUGE } from '../../utils/constants';

const styles = StyleSheet.create({
    heading: {
        color: COLOR_FOREGROUND,
        fontWeight: 'bold',
        textAlign: 'left',
        fontSize: normalizeFontSize(FONT_SIZE_LARGE),
    },
    paragraph: {
        color: COLOR_FOREGROUND,
        textAlign: 'left',
        fontSize: normalizeFontSize(FONT_SIZE),
    },
    label: {
        color: COLOR_FOREGROUND,
        textAlign: 'left',
        paddingHorizontal: normalizeFontSize(5),
        paddingVertical: normalizeFontSize(2),
        fontSize: normalizeFontSize(FONT_SIZE_SMALL),
    },
    value: {
        color: COLOR_FOREGROUND,
        textAlign: 'center',
        textAlignVertical: 'center',
        fontFamily: FONT_FAMILY,
        padding: normalizeFontSize(2),
        fontSize: normalizeFontSize(FONT_SIZE_HUGE),
    },
    button: {
        backgroundColor: COLOR_BACKGROUND,
        color: COLOR_FOREGROUND,
        textAlign: 'center',
        textAlignVertical: 'center',
        alignSelf: 'baseline',
        fontSize: normalizeFontSize(FONT_SIZE),
    },
});

export default styles;
