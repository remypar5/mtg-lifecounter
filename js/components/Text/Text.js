import React from 'react';
import { Text as NativeText } from 'react-native';
import PropTypes from 'prop-types';

import styles from './styles';

const Text = (props) => {
    const { type, style, children } = props;
    return (<NativeText {...props} style={[styles[type], style]}>{children}</NativeText>);
};

Text.propTypes = {
    children: PropTypes.node.isRequired,
    type: PropTypes.oneOf([
        'heading',
        'paragraph',
        'label',
        'value',
        'button',
    ]),
    style: PropTypes.oneOfType([
        PropTypes.object,
        PropTypes.number,
        PropTypes.arrayOf(PropTypes.oneOfType([
            PropTypes.object,
            PropTypes.number,
        ])),
    ]),
};

Text.defaultProps = {
    type: 'paragraph',
    style: null,
};

export default Text;
