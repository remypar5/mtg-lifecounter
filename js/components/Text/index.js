import Text from './Text';
import styles from './styles';

export default Text;
export {
    styles,
};
