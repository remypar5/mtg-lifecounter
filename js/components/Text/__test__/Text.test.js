import React from 'react';
import { shallow } from 'enzyme';

import Text from '../Text';

describe('<Text />', () => {
    let component;

    beforeEach(() => {
        component = shallow(<Text type="heading">Heading</Text>);
    });

    it('renders without crashing', () => {
        expect(component).toMatchSnapshot();
    });
});
