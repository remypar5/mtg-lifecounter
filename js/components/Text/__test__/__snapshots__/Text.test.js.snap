// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Text
    style={null}
    type="heading"
  >
    Heading
  </Text>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "accessible": true,
      "allowFontScaling": true,
      "children": "Heading",
      "ellipsizeMode": "tail",
      "style": Object {
        "color": "#b2b2b0",
        "fontSize": 225,
        "fontWeight": "bold",
        "textAlign": "left",
      },
      "type": "heading",
    },
    "ref": null,
    "rendered": "Heading",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessible": true,
        "allowFontScaling": true,
        "children": "Heading",
        "ellipsizeMode": "tail",
        "style": Object {
          "color": "#b2b2b0",
          "fontSize": 225,
          "fontWeight": "bold",
          "textAlign": "left",
        },
        "type": "heading",
      },
      "ref": null,
      "rendered": "Heading",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
